contracts/modules/staking/StakingPool.sol

```diff
@@ -1544,11 +1557,9 @@ function setProducts(StakedProductParam[] memory params) external onlyManager {
     uint initialCapacityUsed,
     uint totalCapacity
   ) public pure returns (uint) {
-
-    // base price has 18 decimals
     // cover amount has 2 decimals (100 = 1 unit)
-    // dividing by ALLOCATION_UNITS_PER_NXM (=100) to get the right amount of decimals
-    uint basePremium = basePrice * coverAmount / ALLOCATION_UNITS_PER_NXM;
+    // scale coverAmount to 18 decimals and apply price percentage
+    uint basePremium = (coverAmount * NXM_PER_ALLOCATION_UNIT) * basePrice / TARGET_PRICE_DENOMINATOR;
```

Comment:
if basePrice has 18 decimals

coverAmount has 2 decimals + 16 from NXM_PER_ALLOCATION_UNIT

thats' a total of 36.

the division is then by TARGET_PRICE_DENOMINATOR = 100_00

so the final basePremium is off in terms of scaling.

Reply:
In this other comment is the context for this change https://github.com/NexusMutual/smart-contracts/pull/495#discussion_r1024463714